model/device.go

```diff
@@ -183,6 +185,26 @@ func (gn GroupName) String() string {
 	return string(gn)
 }
 
+func (gn GroupName) Validate() error {
```

Comment:
I have a vision of validation go-lib-micro module:
```go
 if validation.AlphaNum(gn) { ... }
```
or even `validation.ValidGroupName(gn)`

Reply:
The `ozzo-validation` package has an interface called `Validateable` that, if implemented, is called whenever validation is required for a field of the given type. I think that's pretty neat :)